n2y/plugins/footnotes.py

```diff
@@ -0,0 +1,106 @@
+import re
+import logging
+
+from pandoc.types import Note, Str, Para
+
+from n2y.rich_text import TextRichText
+from n2y.blocks import ParagraphBlock
+from n2y.errors import UseNextClass
+
+
+plugin_data_key = "n2y.plugins.footnotes"
+
+logger = logging.getLogger(__name__)
+
+
+class ParagraphWithFootnoteBlock(ParagraphBlock):
+    def __init__(self, client, notion_data, page, get_children=True):
+        super().__init__(client, notion_data, page, get_children)
+        if self._is_footnote():
+            self._attach_footnote_data()
+        else:
+            raise UseNextClass()
+
+    def to_pandoc(self):
+        return None
+
+    def _attach_footnote_data(self):
+        if plugin_data_key not in self.page.plugin_data:
+            self.page.plugin_data[plugin_data_key] = {}
+        if self._footnote() not in self.page.plugin_data[plugin_data_key]:
+            self.page.plugin_data[plugin_data_key][self._footnote()] = self._footnote_ast()
+            if self._footnote_ast_empty():
+                logger.warning('Empty footnote "[%s]"', self._footnote())
+        else:
+            msg = 'Multiple footnotes for "[%s]". Skipping latest (%s)'
```

Comment:
I think we haven't been using sentences for these (although I may be wrong)

```suggestion
            msg = 'Multiple footnotes for "[%s]", skipping latest (%s)'
```